### PR TITLE
Make GetInfo queries consistent and tested

### LINKED ISSUE #549: Add pointer validation rules to `GetInfo` entry-points
GetInfo queries have optional parameters for the `propValue` and `pPropSizeRet`, however they still have invalid usage as a combination of other parameters. We should ensure that all GetInfo query entry points validate the input parameters like this:

```cpp
if (propSize == 0 && pPropValue != nullptr) {
  return UR_RESULT_ERROR_INVALID_SIZE;
}
if (propSize != 0 && pPropValue == nullptr) {
  return UR_RESULT_ERROR_INVALID_NULl_POINTER;
}
if (pPropValue == nullptr && pPropSizeRet == nullptr) {
  return UR_RESULT_ERROR_INVALID_NULl_POINTER;
}
```

### PR DESCRIPTION
**Pull request by veselypeta**
Closes #549

### CODE REVIEW

File: scripts/core/kernel.yml [hunk 1]
```diff
@@ -190,6 +190,18 @@ params:
       name: "pPropSizeRet"
       desc: |
             [out][optional] pointer to the actual size in bytes of data being queried by propName.
+returns:
+    - $X_RESULT_ERROR_UNSUPPORTED_ENUMERATION:
+        - "If `propName` is not supported by the adapter."
```
Comment (aarongreig) [inline scripts/core/kernel.yml:195]: this line seems one indentation too far in (here and a few other files)

**Review by aarongreig — APPROVED** [review]
LGTM

File: scripts/core/kernel.yml (hunk 1 above)
Comment (kbenzie) [reply scripts/core/kernel.yml:195]: Tabs are 4 spaces in the project, its all the other ones that don't have the correct intendation.

Comment (veselypeta) [reply scripts/core/kernel.yml:195]: Indented to 4-spaces 👍

**Review by pbalcer — APPROVED** [review]

**Pull request merged**